app/pods/publications/publication/translations/index/controller.js

```diff
@@ -0,0 +1,193 @@
+    const dutch = yield this.store.findRecordByUri('language', CONSTANTS.LANGUAGES.NL);
+    yield Promise.all(uploadedPieces.map((piece) => {
+      piece.translationSubcaseSourceFor = this.translationSubcase;
+      piece.language = dutch;
+      return piece.save();
+    }));
+
+    const requestActivity = this.store.createRecord('request-activity', {
+      startDate: now,
+      translationSubcase: this.translationSubcase,
+      usedPieces: uploadedPieces,
+    });
+    yield requestActivity.save();
+
+    const translationActivity = this.store.createRecord(
+      'translation-activity',
+      {
+        startDate: now,
+        dueDate: translationRequest.translationDueDate,
+        title: translationRequest.subject,
+        subcase: this.translationSubcase,
+        requestActivity: requestActivity,
+        usedPieces: uploadedPieces,
+        language: yield this.store.findRecordByUri(
+          'language',
+          CONSTANTS.LANGUAGES.FR
+        ),
+      }
+    );
+    yield translationActivity.save();
+
+    this.translationSubcase.dueDate = translationRequest.translationDueDate;
+    if (this.translationSubcase.hasDirtyAttributes) {
+      yield this.translationSubcase.save();
+    }
+
+    const [files, outbox, mailSettings] = yield Promise.all([
+      Promise.all(uploadedPieces.mapBy('file')),
+      this.store.findRecordByUri('mail-folder', PUBLICATION_EMAIL.OUTBOX),
+      this.store.queryOne('email-notification-setting'),
+    ]);
+    const mail = this.store.createRecord('email', {
+      to: mailSettings.translationRequestToEmail,
+      from: mailSettings.defaultFromEmail,
+      folder: outbox,
+      attachments: files,
+      requestActivity: requestActivity,
+      subject: translationRequest.subject,
+      message: translationRequest.message,
+    });
+    yield mail.save();
+
+    // PUBLICATION-STATUS
+    yield this.publicationService.updatePublicationStatus(
+      this.publicationFlow,
+      CONSTANTS.PUBLICATION_STATUSES.TO_TRANSLATIONS
+    );
+
+    this.send('refresh');
+    this.showTranslationRequestModal = false;
+  }
+
+  @dropTask
+  *deleteRequest(requestActivity) {
+    const deletePromises = [];
+
+    const translationActivity = yield requestActivity.translationActivity;
+    deletePromises.push(translationActivity.destroyRecord());
+
+    const mail = yield requestActivity.email;
+    if (mail) {
+      deletePromises.push(mail.destroyRecord());
+    }
+
+    deletePromises.push(requestActivity.destroyRecord());
+
+    const pieces = yield requestActivity.usedPieces;
+
+    for (const piece of pieces.toArray()) {
+      const [file, documentContainer] = yield Promise.all([
+        piece.file,
+        piece.documentContainer,
+      ]);
+
+      deletePromises.push(piece.destroyRecord());
+      deletePromises.push(file.destroyRecord());
+      deletePromises.push(documentContainer.destroyRecord());
+    }
+    yield Promise.all(deletePromises);
+    this.send('refresh');
+  }
```

Comment:
code gooit errors, mogelijk omdat ik een aanvraag heb kunnen maken zonder alles in te vullen (enkel doc geupload en aanvraag gestart)